frontend/src/lib/components/button/base/ButtonBase.svelte

```diff
@@ -0,0 +1,49 @@
+<script lang="ts">
+  import type {ButtonBaseProps} from './ButtonBase.type';
+
+  type $$Props = ButtonBaseProps;
+
+  // Element type
+  const tagName = $$restProps['href'] ? 'a' : 'button';
+
+  // Merge classes into $$restProps
+  $$restProps.class = `btn min-h-touch min-w-touch ${$$restProps.class ?? ''}`;
+</script>
+
+<!--
+@component
+A base component for buttons. Consider using the other button components
+in this module instead of this base component.
+
+The item is rendered as an `<a>` element if `href` is supplied. Otherwise
+a `<button>` element will be used. Be sure to provde an `on:click` event 
+handler or other way of making the item interactive.
+
+### Slots
+
+- default: The content of the button.
+
+### Properties
+
+- `href`: The URL to navigate to. If this is not supplied be sure to
+  provde an `on:click` event handler or other way of making the item
+  interactive.
+- Any valid attributes of either an `<a>` or `<button>` element depending
+  whether `href` was defined or not, respectively.
+
+### Usage
+
+```tsx
+<ButtonBase href="/info" icon="info">
+  <Icon name="info"/> 
+  Show info
```

Comment:
is this accurate? do we need to pass the `<Icon>`  and the `icon` attribute?

Reply:
No, that was a typo 😄